Remotion/Data/DomainObjects/Persistence/Rdbms/SqlServer/DbCommandBuilders/SqlDbCommandBuilderFactory.cs

```diff
@@ -72,12 +78,20 @@ public SqlDbCommandBuilderFactory (ISqlDialect sqlDialect)
       ArgumentUtility.CheckNotNull("selectedColumns", selectedColumns);
       ArgumentUtility.CheckNotNull("orderedColumns", orderedColumns);
 
-      var comparedValues = GetValuesForSingleColumnDefinition(comparedColumnValueTable);
+      var (columnDefinition, values) = GetValuesForSingleColumnDefinition(comparedColumnValueTable);
+      var tableType = (TableTypeDefinition)_tableTypeDefinitionProvider.GetStructuredTypeDefinition(columnDefinition.StorageTypeInfo.DotNetType, false);
+
+      var recordDefinition = new RecordDefinition(
+          columnDefinition.Name,
+          tableType,
+          [RecordPropertyDefinition.ScalarAsValue(tableType.Properties.Single())]);
```

Comment:
The fact that there can only be a single property in tableType.Properties is ensured by how we call GetStructuredTypeDefinition, correct?

Reply:
Correct. The `columnDefinition.StorageTypeInfo.DotNetType` must be a scalar type, so the returned `TableTypeDefinition` cannot be anything else than a table type with a single column for that type.
However, a twisted mind could hypothetically replace the `ISingleScalarStructuredTypeDefinitionProvider` with an implementation that returns something else. If there are more than one `IRdbmsStoragePropertyDefinition`, `.Single()` in line 87 would throw; if it is not a `SimpleStoragePropertyDefinition`, `SqlTableValuedParameterValue.AddRecord()` will throw an `ArgumentException`.

Reply:
ah, I see the tableProvider is interface ISingle... 

Do you want to add `Assertion.IsTrue(tableType.Properties.Count == 1, "Table type returned by ISingleScalarStructuredTypeDefinitionProvider should only contain a single property.")` to make it more explicit to better document our contract?